### PR TITLE
CN0561 example

### PR DESCRIPTION
**Pull request by adrimbarean**
Changes this patch:
 - add support for AD4134 to the AD713x driver;
 - add AD7134/AD4134 IIO driver;
 - add cn0561 support for Cora Z7 and Zed board carriers.

### CODE REVIEW

**Comment by adrimbarean** [comment]
V1:
 - fix astyle

File: drivers/adc/ad713x/ad713x.c [hunk 1]
```diff
@@ -541,6 +541,10 @@ int32_t ad713x_init(struct ad713x_dev **device,
 
 	dev->dev_id = init_param->dev_id;
 
+	ret = ad713x_spi_reg_read(dev, AD713X_REG_CHIP_TYPE, &data);
```
Comment (amiclaus) [inline drivers/adc/ad713x/ad713x.c:545]: what's the point of reading the chip_type register if not ID check is performed?

Comment (adrimbarean) [reply drivers/adc/ad713x/ad713x.c:545]: Good point. Will implement.

File: projects/cn0561/builds.json [hunk 2]
```diff
@@ -0,0 +1,12 @@
+{
+  "xilinx": {
```
Comment (amiclaus) [inline projects/cn0561/builds.json:2]: we might need xsa files for this to work with the projects build

File: projects/cn0561/builds.json [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+    "iio": {
+      "flags" : "TINYIIOD=y",
+      "hardware" : ["ad7134_fmc_zed"]
+    },
+    "uart": {
```
Comment (amiclaus) [inline projects/cn0561/builds.json:7]: also, as far as I know we have support both for zed and cora, therefore we can extend this json.

Comment (adrimbarean) [reply projects/cn0561/builds.json:7]: Will do.

File: projects/cn0561/builds.json (hunk 2 above)
Comment (adrimbarean) [reply projects/cn0561/builds.json:2]: Will provide.

**Comment by adrimbarean** [comment]
V2:
- perform chip type check after reading it;
- update builds.json to do all 4 currently supported builds.

File: projects/cn0561/builds.json [hunk 4]
```diff
@@ -0,0 +1,20 @@
+{
+  "xilinx": {
+    "iio_zed": {
+      "flags" : "TINYIIOD=y",
+      "hardware" : ["cn0561_fmc_zed"]
+    },
+    "uart_zed": {
+      "flags" : "",
+      "hardware" : ["cn0561_fmc_zed"]
+    }
```
Comment (amiclaus) [inline projects/cn0561/builds.json:10]: nit: missing comma

File: projects/cn0561/src.mk [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+	$(PLATFORM_DRIVERS)/$(PLATFORM)_gpio.c \
+	$(PLATFORM_DRIVERS)/$(PLATFORM)_spi.c \
+	$(PLATFORM_DRIVERS)/delay.c
+ifeq (y,$(strip $(TINYIIOD)))
+LIBRARIES += iio
+SRC_DIRS += $(NO-OS)/iio/iio_app
+SRCS += $(PLATFORM_DRIVERS)/no_os_uart.c \
+	$(PLATFORM_DRIVERS)/$(PLATFORM)_irq.c \
+	$(DRIVERS)/api/no_os_irq.c \
+	$(NO-OS)/util/no_os_fifo.c \
```
Comment (amiclaus) [inline projects/cn0561/src.mk:32]: missing lf256fifo header and source file.

**Comment by adrimbarean** [comment]
V3:
 - add missing comma to json config;
 - correct cora hardware file name;
 - add custom defines for different platform builds in json config;
 - add lf256fifo files to make import.

**Comment by adrimbarean** [comment]
V4:
 - fix zed build;
 - fix iio driver warnings of unused variables.

**Review by amiclaus — APPROVED** [review]

**Pull request merged**